pkg/cloudprovider/aws/ltresolver/resolver.go

```diff
@@ -0,0 +1,132 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ltresolver
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ssm/ssmiface"
+	"github.com/patrickmn/go-cache"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/cloudprovider"
+	"github.com/aws/karpenter/pkg/cloudprovider/aws/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/cloudprovider/aws/ltresolver/bootstrap"
+)
+
+var defaultEBS = v1alpha1.BlockDevice{
+	Encrypted:  aws.Bool(true),
+	VolumeType: aws.String(ec2.VolumeTypeGp3),
+	VolumeSize: resource.NewScaledQuantity(20, resource.Giga),
+}
+
+// Resolver is able to fill-in dynamic launch template parameters
+type Resolver struct {
+	amiProvider *AMIProvider
+}
+
+// Options define the static launch template parameters
+type Options struct {
+	ClusterName             string
+	ClusterEndpoint         string
+	AWSENILimitedPodDensity bool
+	InstanceProfile         string
+	KubernetesVersion       string
+	CABundle                *string `hash:"ignore"`
+	// Level-triggered fields that may change out of sync.
+	SecurityGroupsIDs []string
+	Tags              map[string]string
+	Labels            map[string]string `hash:"ignore"`
+}
+
+// ResolvedTemplate holds the dynamically generated launch tempalte parameters
+type ResolvedTemplate struct {
+	*Options
+	UserData            bootstrap.Bootstrapper
+	BlockDeviceMappings []*v1alpha1.BlockDeviceMapping
+	MetadataOptions     *v1alpha1.MetadataOptions
+	AMIID               string
+	InstanceTypes       []cloudprovider.InstanceType `hash:"ignore"`
```

Comment:
correct, they are set in the instance provider (within the fleet request). They're in the resolved template because they're kind of a dynamic parameter that is dependent on the AMI selected.

Reply:
I'm super confused by this astraction.